The ipam.spidersubnet in the chart was changed to ipam.spiderSubnet, causing the upgrade to fail

action url: https://github.com/spidernet-io/spiderpool/actions/runs/10799804934

```
ALL_IMAGES: ghcr.io/k8snetworkplumbingwg/multus-cni:v3.9.3 ghcr.io/spidernet-io/spiderpool/spiderpool-plugins:82659d90cae0d6a5169eac2869e47c989932d775 spiderpool-agent-race:main spiderpool-controller-race:main  
ghcr.io/k8snetworkplumbingwg/multus-cni:v3.9.3
Image: "ghcr.io/k8snetworkplumbingwg/multus-cni:v3.9.3" with ID "sha256:c0e8690ae66a1855684cfe1295b3916d63a28918a78cf483eb6ad9dd018da415" found to be already present on all nodes.
ghcr.io/spidernet-io/spiderpool/spiderpool-plugins:82659d90cae0d6a5169eac2869e47c989932d775
Image: "ghcr.io/spidernet-io/spiderpool/spiderpool-plugins:82659d90cae0d6a5169eac2869e47c989932d775" with ID "sha256:ec4a0891f85ae46d1defbaf8baecd6b6ae03c210b0124b9eb648cc035d69b072" found to be already present on all nodes.
spiderpool-agent-race:main
Image: "spiderpool-agent-race:main" with ID "sha256:31ca59e3f1724845ec940e29177067b8bb354b946129d4f540916710dce16fc3" not yet present on node "spiderpool0910202609-control-plane", loading...
Image: "spiderpool-agent-race:main" with ID "sha256:31ca59e3f1724845ec940e29177067b8bb354b946129d4f540916710dce16fc3" not yet present on node "spiderpool0910202609-worker", loading...
spiderpool-controller-race:main
Image: "spiderpool-controller-race:main" with ID "sha256:b2e7e9e06a38bda1c[214](https://github.com/spidernet-io/spiderpool/actions/runs/10799804934/job/29956732433#step:17:215)84b0d5513d5cde264c8aa0799c1edc8cc6d1068e2745" not yet present on node "spiderpool0910202609-control-plane", loading...
Image: "spiderpool-controller-race:main" with ID "sha256:b2e7e9e06a38bda1c21484b0d5513d5cde264c8aa0799c1edc8cc6d1068e2745" not yet present on node "spiderpool0910202609-worker", loading...
upgrade spiderpool with image spiderpool-agent-race:main and spiderpool-controller-race:main 
+ helm --kubeconfig /home/runner/work/spiderpool/spiderpool/test/.cluster/spiderpool0910202609/.kube/config upgrade spiderpool /home/runner/work/spiderpool/spiderpool/charts/spiderpool --set spiderpoolController.replicas=1 --set multus.multusCNI.defaultCniCRName=macvlan-vlan0 --set spiderpoolAgent.image.registry= --set spiderpoolAgent.image.repository=spiderpool-agent-race --set spiderpoolAgent.image.tag=main --set spiderpoolController.image.registry= --set spiderpoolController.image.repository=spiderpool-controller-race --set spiderpoolController.image.tag=main --set spiderpoolInit.image.registry= --set spiderpoolInit.image.repository=spiderpool-controller-race --set spiderpoolInit.image.tag=main --set multus.multusCNI.uninstall=false -n kube-system --debug --reuse-values
upgrade.go:158: [debug] preparing upgrade for spiderpool
upgrade.go:550: [debug] reusing the old release's values
Error: UPGRADE FAILED: template: spiderpool/templates/pod.yaml:60:22: executing "spiderpool/templates/pod.yaml" at <.Values.ipam.spiderSubnet.enable>: nil pointer evaluating interface {}.enable
helm.go:84: [debug] template: spiderpool/templates/pod.yaml:60:22: executing "spiderpool/templates/pod.yaml" at <.Values.ipam.spiderSubnet.enable>: nil pointer evaluating interface {}.enable
UPGRADE FAILED
main.newUpgradeCmd.func2
	helm.sh/helm/v3/cmd/helm/upgrade.go:240
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.8.0/command.go:983
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.8.0/command.go:1115
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.8.0/command.go:1039
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
	runtime/proc.go:271
runtime.goexit
	runtime/asm_amd64.s:1695
make[2]: *** [Makefile:429: helm_upgrade_spiderpool] Error 1
make[2]: Leaving directory '/home/runner/work/spiderpool/spiderpool/test'
make[1]: *** [Makefile:424: upgrade_spiderpool] Error 2
make[1]: Leaving directory '/home/runner/work/spiderpool/spiderpool/test'
make: *** [Makefile:437: upgrade_e2e_spiderpool] Error 2
failed to upgrade spiderpool from v1.0.0-rc1 to main
```

--- a/.github/workflows/auto-upgrade-ci.yaml
+++ b/.github/workflows/auto-upgrade-ci.yaml
@@ -315,7 +315,6 @@ jobs:
 
       - name: Upgrade to version ${{ needs.get_ref.outputs.new_version }}
         id: upgrade
-        continue-on-error: true
         run: |
           RESULT=0
           make upgrade_e2e_spiderpool -e E2E_CLUSTER_NAME=${{ env.E2E_CLUSTER_NAME }} \

--- a/Makefile
+++ b/Makefile
@@ -434,7 +434,7 @@ clean_e2e_spiderpool:
 
 .PHONY: upgrade_e2e_spiderpool
 upgrade_e2e_spiderpool:
-	$(QUIET) make -C test upgrade_spiderpool
+	$(QUIET) make -C test helm_upgrade_spiderpool
 
 .PHONY: codegen
 codegen:

--- a/docs/usage/install/upgrade-zh_CN.md
+++ b/docs/usage/install/upgrade-zh_CN.md
@@ -153,6 +153,10 @@ kubectl patch sp ${auto-pool} --type merge --patch '{"metadata": {"labels": {"ip
 
 在 0.9.4 以下的版本中，statefulSet 应用在快速扩缩容场景下，Spiderpool GC 可能会错误的回收掉 IPPool 中的 IP 地址，导致同一个 IP 被分配给 K8S 集群的多个 Pod，从而出现 IP 地址冲突。该问题已修复，参考[修复](https://github.com/spidernet-io/spiderpool/pull/3778)，但在升级后，冲突的 IP 地址并不能自动被 Spiderpool 纠正回来，您需要通过手动重启冲突 IP 的 Pod 来辅助解决，在新版本中不会再出现错误 GC IP 而导致 IP 冲突的问题。
 
+### 低于 0.9.5 (不包含 0.9.5) 升级到最高版本的注意事项
+
+在 0.9.5 以下的版本中，Spiderpool Charts values.yaml 中的 spiderSubnet 字段发生变更，由 `ipam.spidersubnet` 变更为 `ipam.spiderSubnet`，因此，您无法可以安全地使用 `--reuse-values` 标志从 0.9.5 以下版本升级到 0.9.5 及以上版本。请修改 values.yaml 文件，或者使用 `--set ipam.spiderSubnet.enable=true` 标志来覆盖 values.yaml 文件中的值。
+
 ### 更多版本升级的注意事项
 
 *TODO.*

--- a/docs/usage/install/upgrade.md
+++ b/docs/usage/install/upgrade.md
@@ -153,6 +153,10 @@ Due to the addition of the `txQueueLen` field to the [SpiderCoordinator CRD](./.
 
 In versions below 0.9.4, when statefulSet is rapidly scaling up or down, Spiderpool GC may mistakenly reclaim IP addresses in IPPool, causing the same IP to be assigned to multiple Pods in the K8S cluster, resulting in IP address conflicts. This issue has been fixed, see [Fix](https://github.com/spidernet-io/spiderpool/pull/3778), but after the upgrade, the conflicting IP addresses cannot be automatically corrected by Spiderpool. You need to manually restart the Pod with the conflicting IP to assist in resolving the issue. In the new version, there will no longer be an issue with IP conflicts caused by incorrect GC IPs.
 
+### Upgrading from a version below 0.9.5 (Excludes 0.9.5) to a higher version
+
+In versions lower than 0.9.5, the spiderSubnet field in Spiderpool Charts values.yaml changed from `ipam.spidersubnet` to `ipam.spiderSubnet`, so you cannot safely use the `--reuse-values` flag to upgrade from versions < 0.9.5 to 0.9.5 and above. Please modify the values.yaml file or use the `--set ipam.spiderSubnet.enable=true` flag to override the value in the values.yaml file.
+
 ### More notes on version upgrades
 
 *TODO.*

--- a/test/Makefile
+++ b/test/Makefile
@@ -418,11 +418,6 @@ uninstall_spiderpool:
 			done ; \
 		done
 
-.PHONY: upgrade_spiderpool
-upgrade_spiderpool:
-	@echo -e "\033[35m [upgrade spiderpool] \033[0m"
-	@make helm_upgrade_spiderpool
-
 .PHONY: helm_upgrade_spiderpool
 helm_upgrade_spiderpool:
 	@echo -e "\033[35m [helm upgrade spiderpool] \033[0m"
@@ -442,7 +437,16 @@ helm_upgrade_spiderpool:
 	--set spiderpoolInit.image.registry="" \
 	--set spiderpoolInit.image.repository=$(SPIDERPOOL_CONTROLLER_IMAGE_NAME) \
 	--set spiderpoolInit.image.tag=$(E2E_SPIDERPOOL_TAG) \
-	--set multus.multusCNI.uninstall=false " \
+	--set multus.multusCNI.uninstall=false " ; \
+	if [ "$(E2E_SPIDERPOOL_ENABLE_SUBNET)" == "true" ] ; then \
+		HELM_OPTION+=" --set ipam.spiderSubnet.enable=true " ; \
+		HELM_OPTION+=" --set ipam.spiderSubnet.autoPool.enable=true " ; \
+		HELM_OPTION+=" --set ipam.spiderSubnet.autoPool.defaultRedundantIPNumber=1 " ; \
+	else \
+		HELM_OPTION+=" --set ipam.spiderSubnet.enable=false " ; \
+		HELM_OPTION+=" --set ipam.spiderSubnet.autoPool.enable=true " ; \
+		HELM_OPTION+=" --set ipam.spiderSubnet.autoPool.defaultRedundantIPNumber=1 " ; \
+	fi ; \
 	ALL_IMAGES=`helm template $(RELEASE_NAME) $(ROOT_DIR)/charts/spiderpool $${HELM_OPTION} | grep ' image: ' | tr -d '"' | awk -F 'image: ' '{print $$2}' | sort | uniq | tr '\n' ' '` ; \
 	echo "ALL_IMAGES: $${ALL_IMAGES} " ; \
 	for IMAGE in $${ALL_IMAGES}; do \

